_hardware/cypherockx1.md

```diff
@@ -35,33 +35,72 @@ social:
 
 ## Interface
 
-The device has a screen and one circular button that allows users to select the letters. It is quite tedious as the single button is very limited in the number of letters it can select with one push of this button. This is evident in the [video shown here](https://youtu.be/eD3CZMPumCk?t=204).
+The device has a screen and one circular button that allows users to select the letters. It is quite tedious as the single button is very limited in the number of letters it can select with one push of this button. This is evident in the [video shown here](https://www.youtube.com/watch?v=VadX0ERgEn0).
 
-## Reproducibility
+## Public repository containing source code:
+> - Cypherock X1 wallet firmware is open sourced, hosted in GitHub repo: [Cypherock/x1_wallet_firmware](https://github.com/Cypherock/x1_wallet_firmware)
+> - Latest release source code: [Release v0.4.773](https://github.com/Cypherock/x1_wallet_firmware/releases/tag/v0.4.773)
+> - Cypherock X1 wallet firmware is released in 2 flavors, “***initial firmware***” and “***main firmware***”.
+    > - ***Initial firmware*** is installed on the device during manufacturing. It provides minimum functionality such as device authentication and card authentication which checks the genuinity of the product. This firmware does not support any wallet related functionality. The initial firmware ceases to exist before any wallet activity is initiated.
+   > - ***Main firmware provides*** full fledged wallet related functionality and is installed on the device via the companion Cypherock CySync desktop application.
+## Fingerprint of the latest release binary: 
+> - Fingerprint (SHA256 digest) of the latest Cypherock X1 wallet firmware binary (unsigned) can be found on the release page: [Release v0.4.773](https://github.com/Cypherock/x1_wallet_firmware/releases/tag/v0.4.773)
 
-In its [2020 Bitcointalk.org ANN post](https://bitcointalk.org/index.php?topic=5225800.0), Cypherock claims that the hardware wallet **will** be open source. It also makes this claim on [its website](https://twitter.com/BitcoinWalletz/status/1462700172719243266). The product is still in pre-order. We were able to locate Cypherock's Github page, but there was [no corresponding repository](https://twitter.com/BitcoinWalletz/status/1462694293085118466) specifically for its hardware wallet. 
+## Reproducibility of the release binary from the source code
+> - Cypherock X1 wallet firmware binary is signed by two private keys. The signature and a signed header is appended on top of firmware binary to form the complete Cypherock X1 wallet firmware image. 
+> - An important thing to note here is that the Cypherock X1 wallet firmware release binary is compiled using Docker, so, the binary will be reproducible only if the environment settings match. 
```

Comment:
Hi Giszmo, the intention of the line was same as you mentioned. We use Docker to improve the consistency of builds. 
If anyone wants to reproduce the release binaries, they must create the same environment. 

I will reword this point so that it does not mislead anyone to think otherwise.